packages/app/src/server/routes/api/cortex/index.ts

```diff
@@ -31,6 +32,33 @@ export const cortexProxy = httpProxy.createProxyServer({ ignorePath: true });
  */
 cortexProxy.on("proxyReq", onProxyReq);
 
+cortexProxy.on("proxyRes", (proxyRes, req, res) => {
+  const isHTMLResponse = proxyRes.headers["content-type"]?.includes("text/html")
+  const body: Array<any> = [];
+  res.writeHead(proxyRes.statusCode!, proxyRes.headers);
+  proxyRes.on("data", function (chunk) {
+    if (!isHTMLResponse) {
+      res.write(chunk)
+    }
+    body.push(chunk)
+  });
+  proxyRes.on("end", function () {
+    if (isHTMLResponse) {
+      /* Some error messages are returned via HTML.
+       * We have mainly seen it with configuration errors, e.g. when sharding 
+       * is disabled. 
+       *
+       * For example HTML responses please check the tests.
+       */
+      const parsedHTML = parse(body);
+      const data = parsedHTML.querySelector("p").text;
+      res.end(data);
+    } else {
+      res.end();
```

Comment:
Does this mean we write a response with an empty body?

Reply:
No, because in the case of `!isHTMLResponse` we already stream the response to the client in [proxyRes.on("data")](https://github.com/opstrace/opstrace/pull/900/files#diff-4ba3f8112ae1c4d28b5397690668351095b7568556ad0bce0d6e2bf0b067d4dbR41).